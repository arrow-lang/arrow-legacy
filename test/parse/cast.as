x as bool;
&y as int;
&z as *byte;
