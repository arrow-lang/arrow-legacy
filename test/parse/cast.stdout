{
  "tag": "Module",
  "span": "test/parse/cast.as:1.1-3.12",
  "name": "cast",
  "block": {
    "tag": "Block",
    "span": "test/parse/cast.as:1.1-3.12",
    "statements": [
      {
        "tag": "Cast",
        "span": "test/parse/cast.as:1.1-10",
        "lhs": {
          "tag": "Identifier",
          "span": "test/parse/cast.as:1.1-2",
          "text": "x"
        },
        "rhs": {
          "tag": "TypePath",
          "span": "test/parse/cast.as:1.6-10",
          "segments": [
            "bool"
          ]
        }
      },
      {
        "tag": "Cast",
        "span": "test/parse/cast.as:2.1-10",
        "lhs": {
          "tag": "AddressOf",
          "span": "test/parse/cast.as:2.1-3",
          "mutable": false,
          "operand": {
            "tag": "Identifier",
            "span": "test/parse/cast.as:2.2-3",
            "text": "y"
          }
        },
        "rhs": {
          "tag": "TypePath",
          "span": "test/parse/cast.as:2.7-10",
          "segments": [
            "int"
          ]
        }
      },
      {
        "tag": "Cast",
        "span": "test/parse/cast.as:3.1-12",
        "lhs": {
          "tag": "AddressOf",
          "span": "test/parse/cast.as:3.1-3",
          "mutable": false,
          "operand": {
            "tag": "Identifier",
            "span": "test/parse/cast.as:3.2-3",
            "text": "z"
          }
        },
        "rhs": {
          "tag": "TypePointer",
          "span": "test/parse/cast.as:3.7-12",
          "mutable": false,
          "pointee": {
            "tag": "TypePath",
            "span": "test/parse/cast.as:3.8-12",
            "segments": [
              "byte"
            ]
          }
        }
      }
    ]
  }
}
